### PR TITLE
feat: Stop returning token associations and token balances data in queries as per HIP-367

### LINKED ISSUE #10089: Deprecate usage of queries related to `Unlimited Token Associations Per Account`
Since all queries related to token associations will be moved to be done against mirror node,  we need to deprecate and stop returning token association information in the following queries related to  [HIP-367](https://hips.hedera.com/hip/hip-367)
- getAccountBalance
- getAccountInfo
- getContractInfo

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #10089 

- Stop returning token associations and token balances data in queries as per HIP-367
The queries modified are 
     - GetAccountBalance
     - GetAccountInfo
     - GetContractInfo
- Changes are done in both mono and modular code
- Changed test-clients to do a `getAccountDetails` query for testing if there are any assertions related to token relationships or token balances

### CODE REVIEW

File: hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkGetAccountDetailsHandler.java [hunk 1]
```diff
@@ -196,42 +200,60 @@ private static Optional<AccountDetails> infoForAccount(
...
+        final var ret = new ArrayList<TokenRelationship>();
         var tokenId = account.headTokenId();
         int count = 0;
-
-        while (tokenId != null && !tokenId.equals(TokenID.DEFAULT) && count <= maxRelsPerInfoQuery) {
-            final var tokenRelation = tokenRelationStore.get(account.accountId(), tokenId);
+        TokenRelation tokenRelation;
+        Token token; // token from readableToken store by tokenID
+        AccountID accountID; // build from accountNumber
+        while (tokenId != null && !tokenId.equals(TokenID.DEFAULT) && count < maxRelsPerInfoQuery) {
```
Comment (tinker-michaelj) [inline hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkGetAccountDetailsHandler.java:218]: I'm uneasy about patterns where we treat it as possible for `TokenID.DEFAULT` to be used as an alternative to `null` to indicate a missing field.

**Review by tinker-michaelj — DISMISSED** [review]
LGTM, ty @Neeharika-Sompalli !

File: hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkGetAccountDetailsHandler.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkGetAccountDetailsHandler.java]: Removed that. Thanks @tinker-michaelj

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/records/TokenBalanceValidation.java [hunk 2]
```diff
@@ -74,6 +74,7 @@ private TokenBalanceValidation(
      * Set up validator. Public constructor for use from <code>TokenReconciliationValidator</code>>
      * Assumes that the tokens have already been created, and validation is only to check that
      * <code>expectedTokenBalances</code> match values returned by <code>hasTokenBalance</code>
+     * NOTE: Since token balances are no more returned from query, this validator's validation should be changed
```
Comment (mhess-swl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/records/TokenBalanceValidation.java:77]: Changing the validation isn't required as part of this PR?

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @Neeharika-Sompalli !

**Review by mhess-swl — APPROVED** [review]

**Pull request merged**